src/backends/beamcoder.ts

```diff
@@ -282,7 +282,7 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
         // seek and create a decoder when retrieving a frame for the first time or when seeking backwards
         // we have to create a new decoder when seeking backwards as the decoder can only process frames in
         // chronological order.
-        if (!this.#previousTargetPTS || this.#previousTargetPTS > targetPTS) {
+        if (this.#previousTargetPTS === null || this.#previousTargetPTS > targetPTS) {
```

Comment:
oooo sneaky/tricky

Reply:
I guess this only helps when our previousTargetPTS is 0

Reply:
yeah, does this explain why it gets progressively slower though?

Reply:
@michaeljherrmann 

Because we had to parse more packets for every frame.

Made up example to illustrate:


1st frame: seek, read, decode
2nd frame: seek, read, read, decode
3rd frame: seek, read, read, read, decode
....
Nth frame: seek, read, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, readread, read, read, read, read, read, read, read, read, read, read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,read, read,.... decode